internal/auth/auth_test.go

```diff
@@ -0,0 +1,160 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// created from curl https://auth.prod-preview.openshift.io/api/token/keys?format=pem
+var goodKeys = `
+{
+  "keys": [
+    {
+      "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAmCUKKr6LbO9eYKPyJB3kirzlSV8GpgmUcpTYoiDziLAxc2svPotyyUThWafEmCQ6OR46z60hZImaJcxDXyJJRKFjIA5LGA3ERbLdUGwLmJOE1ZZtcF5+c/pWqREsi5+E5VG/WN2I2ZNCqlQ+cfCqTwuQg+7Pr1jJvxm6Xf268r25kKRKD8uc3bRFaZNSmaK+G3KcZSiFyCAPgQcQU8ZN/9bLjGC8i5DQgCU5SYjvs1SmAUpcEkyVA5IQEuks0Wg0ZCogX50Bpx7FMV+Pbh4wE8tgOr/c8CHQAIGeyzRA2bi1FOcnUeA2OEhGKVv6KxCGsGkxrGpNxDXMYTaiVK2yzwIDAQAB",
+      "kid": "zD-57oBFIMUZsAWqUnIsVu_x71VIjd1irGkGUOiTsL8"
+    },
+    {
+      "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA0eiOo86RtUEil7vbUQWW2rXDSNnGqXjXxxSAeCoBv5ecWGH7ZagmoFPfjSfjxT2qdQcTlRMkQzwsrztPbl3nKYkj/SCgxk0BPVQKWKJObe2bYIzxK8a+s2yvcS7LNrKV5+oWrOG6X46KQJbNh3SyIGy1o0lbBiOTNkQl984JgL9iuxJByz5azjU5W36IP6Hi2eJEYZxmve+I6jGF7PGx6sa7vNHcRDdxBU578IFw5CaMq+2zfza2AQwEH4LKb2bnTmbotgYPjhANgSl6oiqWehzFqeyDuZGJZAzkj31NLguNhmd6MbKCx5QmGJB7907cN1slvYcRA8UTKImfaK3+wQIDAQAB",
+      "kid": "PE6-BEECZZpPZIVxLR6NinbthOHJcGqYrfl8v7v6BVA"
+    }
+  ]
+}
+`
+
+var badKeys = `
+{
+  "keys": [
+    {
+      "key": "a bad key",
+      "kid": "zD-bad-keyid"
+    },
+    {
+      "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA0eiOo86RtUEil7vbUQWW2rXDSNnGqXjXxxSAeCoBv5ecWGH7ZagmoFPfjSfjxT2qdQcTlRMkQzwsrztPbl3nKYkj/SCgxk0BPVQKWKJObe2bYIzxK8a+s2yvcS7LNrKV5+oWrOG6X46KQJbNh3SyIGy1o0lbBiOTNkQl984JgL9iuxJByz5azjU5W36IP6Hi2eJEYZxmve+I6jGF7PGx6sa7vNHcRDdxBU578IFw5CaMq+2zfza2AQwEH4LKb2bnTmbotgYPjhANgSl6oiqWehzFqeyDuZGJZAzkj31NLguNhmd6MbKCx5QmGJB7907cN1slvYcRA8UTKImfaK3+wQIDAQAB",
+      "kid": "PE6-BEECZZpPZIVxLR6NinbthOHJcGqYrfl8v7v6BVA"
+    }
+  ]
+}
+`
+
+type fakeAuthService struct {
```

Comment:
mockAuthService?

Reply:
fakeAuth is equally unambiguous but if I have to make changes to the code to address any comments, I will change it mock.